sdk/python/feast/infra/passthrough_provider.py

```diff
@@ -427,3 +439,12 @@ def validate_data_source(
         data_source: DataSource,
     ):
         self.offline_store.validate_data_source(config=config, data_source=data_source)
+
+    def get_table_column_names_and_types_from_data_source(
+        self, config: RepoConfig, data_source: DataSource
+    ) -> Iterable[Tuple[str, str]]:
+        if isinstance(data_source, FileSource):
```

Comment:
why do we need this? note that FileSource is not necessarily local, it might be an S3 path that's only accessible from offline server in a secure deployment.

Reply:
main reason is that ITs are failing, probably due to messages like 
```
  sdk/python/feast/feature_view.py:425: UserWarning: There are some mismatches in your feature view: order registered entities. Please check if you have applied your entities correctly.Entities: ['customer_id', 'driver'] vs Entity Columns: [driver_id-Int64]
```
but if S3 is also there, it must be fixed differently. I will investigate further

Reply:
The root cause of the original issue `RemoteOfflineStoreConfig' object has no attribute 'user'` is the following: in a distributed setup with a remote offline server, `feast apply` cannot run from the client host because some functions assume to be executed using the actual feast `offline_store` configuration, not the remote one.
`DataSource.get_table_column_names_and_types` is one of such functions.

The idea behind this PR was to move the `get_table_column_names_and_types` execution under the offline store responsibilities, so that it could execute within the offline server in case of a distributed setup. 
By passing the data source name to the remote request, the offline server can fetch the data source instance from the registry and delegate execution to it, allowing the data source to run using its actual connection.

This idea fails miserably until the data source has been successfully registered, e.g. `feast apply` was properly executed, which is exactly the issue we want to fix.

So, the only option to let the offline store handle the data source instance without fetching it from the DB seems to pass its entire configuration to the server, maybe using the protos to easily rebuild the instance. WDYT?

Reply:
@tokoko PR ready: using protos instead of names